ReactiveCocoa/Swift/Signal.swift

```diff
@@ -204,25 +219,47 @@ public func filter<T, E>(predicate: T -> Bool) -> Signal<T, E> -> Signal<T, E> {
 					}
 
 				default:
-					observer.put(event)
+					observer(event)
 				}
-			})
+			}
 		}
 	}
 }
 
-/// Unwraps non-`nil` values from `signal` and forwards them on the returned
-/// signal, `nil` values are dropped.
-public func ignoreNil<T, E>(signal: Signal<T?, E>) -> Signal<T, E> {
-	return signal |> filter { $0 != nil } |> map { $0! }
+/// Maps values from `signal` and forwards the non-nil ones on the returned signal.
+public func filterMap<T, U, E>(transform: T -> U?) -> Signal<T, E> -> Signal<U, E> {
+	return { $0.filterMap(transform) }
+}
+
+extension SignalType {
+	public func filterMap<U>(transform: T -> U?) -> Signal<U, E> {
+		return Signal { observer in
+			return self.signal.observe { event in
+				switch event {
+				case let .Next(value):
+					transform(value).flatMap { sendNext(observer, $0) }
+				case let .Error(error):
+					sendError(observer, error)
+				case .Completed:
+					sendCompleted(observer)
+				case .Interrupted:
+					sendInterrupted(observer)
+				}
+			}
+		}
+	}
 }
 
 /// Returns a signal that will yield the first `count` values from the
 /// input signal.
 public func take<T, E>(count: Int) -> Signal<T, E> -> Signal<T, E> {
```

Comment:
Those are needed for `SignalProducer` operator lifting since I haven't updated those yet to use protocol extensions. So far I've only done `Signal`.